flytekit/common/local_workflow.py

```diff
@@ -0,0 +1,371 @@
+from flytekit.common import launch_plan as _launch_plan
+from flytekit.common import nodes as _nodes
+from flytekit.common import promise as _promise
+from flytekit.common.core import identifier as _identifier
+from flytekit.common.exceptions import user as _user_exceptions
+from flytekit.common.types import helpers as _type_helpers
+from flytekit.common.workflow import SdkWorkflow
+from flytekit.configuration import internal as _internal_config
+from flytekit.models import common as _common_models
+from flytekit.models import interface as _interface_models
+from flytekit.models import literals as _literal_models
+from flytekit.models.core import identifier as _identifier_model
+from flytekit.models.core import workflow as _workflow_models
+
+
+# Local-only wrapper around binding data and variables. Note that the Output object used by the end user is a yet
+# another layer on top of this.
+class Output(object):
+    def __init__(self, name, value, sdk_type=None, help=None):
+        """
+        :param Text name:
+        :param T value:
+        :param U sdk_type: If specified, the value provided must cast to this type.  Normally should be an instance of
+            flytekit.common.types.base_sdk_types.FlyteSdkType.  But could also be something like:
+
+            list[flytekit.common.types.base_sdk_types.FlyteSdkType],
+            dict[flytekit.common.types.base_sdk_types.FlyteSdkType,flytekit.common.types.base_sdk_types.FlyteSdkType],
+            (flytekit.common.types.base_sdk_types.FlyteSdkType, flytekit.common.types.base_sdk_types.FlyteSdkType, ...)
+        """
+        if sdk_type is None:
+            # This syntax didn't work for some reason: sdk_type = sdk_type or Output._infer_type(value)
+            sdk_type = Output._infer_type(value)
+        sdk_type = _type_helpers.python_std_to_sdk_type(sdk_type)
+
+        self._binding_data = _interface.BindingData.from_python_std(sdk_type.to_flyte_literal_type(), value)
+        self._var = _interface_models.Variable(sdk_type.to_flyte_literal_type(), help or "")
+        self._name = name
+
+    def rename_and_return_reference(self, new_name):
+        self._name = new_name
+        return self
+
+    @staticmethod
+    def _infer_type(value):
+        # TODO: Infer types
+        raise NotImplementedError(
+            "Currently the SDK cannot infer a workflow output type, so please use the type kwarg "
+            "when instantiating an output."
+        )
+
+    @property
+    def name(self):
+        """
+        :rtype: Text
+        """
+        return self._name
+
+    @property
+    def binding_data(self):
+        """
+        :rtype: flytekit.models.literals.BindingData
+        """
+        return self._binding_data
+
+    @property
+    def var(self):
+        """
+        :rtype: flytekit.models.interface.Variable
+        """
+        return self._var
+
+
+class PythonWorkflow(SdkWorkflow):
+    """
+    Wrapper class for locally defined Python workflows
```

Comment:
what does locally defined mean? Isn't everything locally defined?

Reply:
let me know what name to use?  happy to rename.  SdkRunnableWorkflow?